include/igl/ray_mesh_intersect.cpp

```diff
@@ -62,21 +90,32 @@ template <
   typename DerivedV,
   typename DerivedF>
 IGL_INLINE bool igl::ray_mesh_intersect(
-  const Eigen::MatrixBase<Derivedsource> & source,
+  const Eigen::MatrixBase<Derivedsource> & s,
   const Eigen::MatrixBase<Deriveddir> & dir,
   const Eigen::MatrixBase<DerivedV> & V,
   const Eigen::MatrixBase<DerivedF> & F,
   igl::Hit & hit)
 {
-  std::vector<igl::Hit> hits;
-  ray_mesh_intersect(source,dir,V,F,hits);
-  if(hits.size() > 0)
+  if (F.rows() == 1)
   {
-    hit = hits.front();
-    return true;
+    // Shortcut for AABB based queries.
```

Comment:
would it make more sense for the AABB code to call this one directly?

Reply:
That is a good question. I haven't really looked at the `AABB` code. Do you think people would use this function here directly, could this be a common use case? If the `AABB` is the only caller with one triangle, guess adapting there would be the better place.